UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/ProductDataService.cs

```diff
@@ -1,128 +1,147 @@
+                ExchangeSetBatchStatusUri = new LinkSetBatchStatusUri { Href = "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272" },
+                ExchangeSetFileUri = new LinkSetFileUri { Href = "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272/files/exchangeset123.zip" }
             };
 
-            await Task.CompletedTask;
-            return exchangeSetResponse;
+            return response;
         }
 
         public Task<ValidationResult> ValidateProductDataByProductVersions(ProductDataProductVersionsRequest request)
         {
             return productVersionsValidator.Validate(request);
         }
 
-        public async Task<ExchangeSetResponse> CreateProductDataSinceDateTime(ProductDataSinceDateTimeRequest productDataSinceDateTimeRequest)
+        public async Task<ExchangeSetServiceResponse> CreateProductDataSinceDateTime(ProductDataSinceDateTimeRequest productDataSinceDateTimeRequest)
         {
-            ExchangeSetResponse exchangeSetResponse = new ExchangeSetResponse
+            var salesCatalogueResponse = await salesCatalogueService.GetProductsFromSpecificDateAsync(productDataSinceDateTimeRequest.SinceDateTime);
+            ////can check for file size from salesCatalogueResponse
+            var response = SetExchangeSetResponse(salesCatalogueResponse, false);
+            if (response.HttpstatusCode != HttpStatusCode.OK)
             {
-                ExchangeSetCellCount = 15,
-                ExchangeSetUrlExpiryDateTime = Convert.ToDateTime("2021-02-17T16:19:32.269Z").ToUniversalTime(),
-                RequestedProductCount = 22,
-                RequestedProductsAlreadyUpToDateCount = 5,
-                RequestedProductsNotInExchangeSet = new List<RequestedProductsNotInExchangeSet>
-                {
-                    new RequestedProductsNotInExchangeSet { ProductName = "GB123456", Reason = "productWithdrawn" },
-                    new RequestedProductsNotInExchangeSet { ProductName = "GB123789", Reason = "invalidProduct" }
-                },
-                Links = new Links()
-                {
-                    ExchangeSetBatchStatusUri = new LinkSetBatchStatusUri { Href = "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272" },
-                    ExchangeSetFileUri = new LinkSetFileUri { Href = "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272/files/exchangeset123.zip" }
-                }
+                return response;
+            }
+            //// FSS call for creating Batch and fill data for _links, exchangeSetUrlExpiryDateTime etc
+            response.ExchangeSetResponse.Links = new Links()
+            {
+                ExchangeSetBatchStatusUri = new LinkSetBatchStatusUri { Href = "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272" },
+                ExchangeSetFileUri = new LinkSetFileUri { Href = "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272/files/exchangeset123.zip" }
```

Comment:
Is there a reason to hard code these Url's into the code or should they be moved into a configuration file?

Reply:
Yes this is been moved to common method and made configuration in PBI 2790 where it actually creates the Batch and based on that those links is generated